datapower-docker-compose-dp-mq/datapower-docker-compose-dp-mq.mkd

```diff
@@ -0,0 +1,168 @@
+
+if [ "$TIMEOUT" -a "$TIMEOUT" -gt 0 ]
+then
+  STOPTIME=$(( $(date +%s) + $TIMEOUT ))
+else
+  STOPTIME=9999999999
+fi
+
+RC=1
+
+while [ $(date +%s) -lt $STOPTIME ]
+do
+  if curl --silent --insecure --form "project=@/usr/local/bin/grow-soapui-project.xml" --form "suite=sendRequestGrow" http://soapui:3000 | grep "200 OK"
+  then
+    RC=0
+    if [ "$CONTINUOUS" != "true" ]
+    then
+      echo "SUCCESS! Exiting"
+      exit $RC
+    fi
+  else
+    echo curl error $?
+  fi
+  sleep 5
+done
+```
+The script curls a SoapUI project named `grow-soapui-project.xml`, located in the /usr/local/bin/ directory, to `src_soapui_1`. The script is set to send a request message every five seconds until a successful transaction is completed. Note that the console log may show a few request failures as the application comes up.
+
+`src_soapui_1` accepts the curl request and processes the test defined in the SoapUI project suite. This is where `bob` is created as an XML message and sent to `src_datapower_1`.
+
+`src_datapower_1` accepts `bob` from `src_soapui_1`. The processing policy configured in the multi-protocol gateway, in the application domain, retrieves various stylesheets and schema from `src_http_1` and transforms `bob`. The policy itself is simple. As `bob` passes through each rule in the policy, his number of nodes and elements increase. At the end, `bob` is a message that reports what size he was in bytes at the end of the trip through DataPower. Bob is then routed to the `src_mq_1` back end.
+
+`bob` is placed on a queue in `src_mq_1` and his travel could very well end here; however, to avoid the queue getting full on a long running test, I set up a dummy service on `src_datapower_1` that clears the queue and throws the message away. 
+
+## Making Changes- You have the power
+The power to make changes in this sample is up to you. The construction of the environment lends itself to customization. The processing policy in the DataPower configuration can be replaced; the SoapUI project can be changed with a different test or additional requests could be added; the curl driver parameters can be adjusted to change the rate and duration of test messages; multiple MQ queue managers can be added. The list is endless and completely up to you.
+
+##Cleaning Up
+When you have completed your testing, you can remove the containers created during the exercise with the following command:
+```
+docker-compose rm
+```
+You can rerun the environment by issuing the docker-compose run command again.
+
+## Conclusion
+So now that `bob's` journey has ended, you can see how an entire docker-composed application can be easily assembled and run. Docker Compose takes care of the assembly of the application and all of the networking within. Customization of the application can easily be implemented to meet your needs. I hope this sample is helpful. Thanks for participating and please provide feedback.
```

Comment:
- Already stopped talking about the transaction, that ended long before cleaning up

I was able to use off-the-shelf Docker images and tools to create a portable containerized application in a very short period of time. I can run it wherever I can run Docker Compose. All the configuration I require is in version control. I can generate relatively complicated requests using SoapUI and I can use DataPower, MQ, and potentially other IBM Middleware as first-class citizens in my containerized applications. And if I can do it, you can too!

I howe you find this sample helpful. I appreciate your time and attention and look forward to your feedback!